### ISSUE TITLE
espnet example?

### ISSUE THREAD

**Issue by danpovey** [lhotse-speech/lhotse#171]
I have been following espnet https://github.com/espnet/espnet and they are open to new recipes, they have merged a couple in the last week or so.  I think lhotse+espnet would be a good combination because Shinji's real interest is, I think, in end-to-end architectures and abstracting away the data-prep may appeal to him.
Would also be a good opportunity to tune things like data augmentation recipes.

**Comment by pzelasko** [comment]
I agree. I talked with Shinji some time ago and I think he was interested to explore that. @sw005320 WDYT? 

Actually I think the discussion in PR #172 could be relevant in the context of ESPnet too.

**Comment by sw005320** [comment]
Yes, this is very cool.
I don't fully follow #172, but #149 would be a good way to use the lhotse data preparation.
@pzelasko, in #149, you mentioned that it was intended to build a Kaldi system.
Then, do you have a Kaldi nsc recipe with lhotse-based data preparation?
Then, I think I can make an nsc recipe in espnet through lhotse-based data preparation.

**Comment by pzelasko** [comment]
Yeah, I have a Kaldi NSC recipe here: https://github.com/pzelasko/kaldi/tree/feature/nsc-recipe/egs/nsc/s5

However, it is not well-tuned - I simply replicated the LibriSpeech recipe as it has a similar amount of data, and got about 31% WER on a held-out, unofficial test set (I don't think there is an official one).

**Comment by sw005320** [comment]
I just made an NSC recipe by using the lhotse data preparation in espnet https://github.com/espnet/espnet/pull/2846
It seems to be working.
I think this is a good practice as an initial point.

**Comment by pzelasko** [comment]
Cool! 

Going forward, would you be interested to explore using Lhotse's Python API? You can see [an example in snowfall's librispeech recipe here](https://github.com/k2-fsa/snowfall/blob/6e198ccbf2dfba2ab0858ee26203425c68045782/egs/librispeech/asr/simple_v1/prepare.py#L75) and then [K2 training uses a dedicated Dataset class here](https://github.com/lhotse-speech/lhotse/blob/master/lhotse/dataset/speech_recognition.py#L17) (I could help with writing the Dataset class, I'm actually trying to figure out what is the right way to make a generic template or a set of utilities for that to support many tasks)

As Dan mentioned that'd definitely help us add and tune relevant features like data augmentation, etc.
